state/unit.go

```diff
@@ -935,7 +936,14 @@ func (u *Unit) removeOps(asserts bson.D, op *ForcedOperation, destroyStorage boo
 		// If we cannot find application, no amount of force will succeed after this point.
 		return nil, err
 	}
-	return app.removeUnitOps(u, asserts, op, destroyStorage)
+	ops, err := app.removeUnitOps(u, asserts, op, destroyStorage)
+	if err != nil {
+		return nil, err
+	}
+	if u.modelType == ModelTypeCAAS {
+		ops = append(ops, removeUnitVirtualHostKeysOps(u.st, u.UnitTag().Id())...)
```

Comment:
Why only for CAAS?

Reply:
Because when adding, we only add virtual host keys for caas units. So the removal side is the same.